pkg/ds/daemon_set.go

```diff
@@ -559,11 +554,10 @@ func (ds *daemonSet) PublishToReplication() error {
 	// InitializeReplicationWithCheck, we will get an error
 	ds.cancelReplication()
 
-	podLocations, err := ds.CurrentPods()
+	nodes, err := ds.EligibleNodes()
```

Comment:
This seems right, and probably more correct than it was before. I suspect that this whole PR will cause PublishToReplication to target a broader set of nodes while narrowing the scope of the other mutative functions (addPods, removePods, clearPods) to be more correct.